src/class/iter.rs

```diff
@@ -104,31 +99,20 @@ where
 {
     #[inline]
     fn tp_iternext() -> Option<ffi::iternextfunc> {
-        py_unary_refmut_func!(
-            PyIterNextProtocol,
-            T::__next__,
-            IterNextConverter::<T::Success>(std::marker::PhantomData)
-        )
+        py_unary_refmut_func!(PyIterNextProtocol, T::__next__, IterNextConverter)
     }
 }
 
-struct IterNextConverter<T>(std::marker::PhantomData<T>);
+struct IterNextConverter<T>(Option<T>);
 
-impl<T> CallbackConverter for IterNextConverter<T>
+impl<T> IntoPyCallbackResult<*mut ffi::PyObject> for IterNextConverter<T>
 where
     T: IntoPy<PyObject>,
 {
-    type Source = Option<T>;
-    type Result = *mut ffi::PyObject;
-    const ERR_VALUE: Self::Result = ptr::null_mut();
-
-    fn convert(val: Self::Source, py: Python) -> Self::Result {
-        match val {
-            Some(val) => val.into_py(py).into_ptr(),
-            None => unsafe {
-                ffi::PyErr_SetNone(ffi::PyExc_StopIteration);
-                ptr::null_mut()
-            },
+    fn convert(self, py: Python) -> PyResult<*mut ffi::PyObject> {
+        match self.0 {
+            Some(val) => Ok(val.into_py(py).into_ptr()),
+            None => Err(crate::exceptions::StopIteration::py_err(())),
```

Comment:
I checked this works almost the same as `ffi::PyErr_SetNone(ffi::PyExc_StopIteration)`

Reply:
👍

Reply:
I made the switch both because it now returns `PyResult` and because I want to reduce the use of `unsafe` as much as possible. In this case I saw we had a safe API to do the same thing, so I chose to use it.